### PR TITLE
[IMP] web: introduce overlay service

### PR DESCRIPTION
**Pull request by mcm-odoo**
This commit adds a new service "overlay" that will be used as a base
for other service which displays a component on foreground like
"popover" or "dialog". The services "popover", "dialog" and "effect"
already uses this new service to have a common container.

The "overlay" service also fix a stacking context issue that could
happen when a popover opened a dialog and this dialog then opened
another popover thanks to the common container.

task id: 3233266

### CODE REVIEW

File: addons/web/static/src/views/fields/properties/properties_field.js [hunk 1]
```diff
@@ -107,43 +107,43 @@ export class PropertiesField extends Component {
 
     /**
      * Return false if we should not close the popover containing the
```
Comment (brboi) [inline addons/web/static/src/views/fields/properties/properties_field.js:109]: Return `true` if we should close...

File: addons/web/static/src/legacy/utils.js [hunk 2]
```diff
@@ -40,9 +45,11 @@ export function makeLegacyRpcService(legacyEnv) {
  */
 export function makeLegacyDialogMappingService(legacyEnv) {
     return {
-        dependencies: ["ui", "hotkey"],
+        dependencies: ["ui", "hotkey", "overlay"],
         start(env) {
-            const { ui, hotkey } = env.services;
+            const { ui, hotkey, overlay } = env.services;
+
+            overlay.add(LegacyDialogContainer, {});
```
Comment (brboi) [inline addons/web/static/src/legacy/utils.js:52]: If I understand correctly, if a legacy dialog and a standard dialog are both open, the legacy one would always be behind.

Is that right ? Do we have cases where this is an issue ?

**Review by brboi — COMMENTED** [review]
I am eager to see this work merged, well done!

**Comment by brboi** [comment]
robodoo r+
\o/

**Comment by brboi** [comment]
robodoo r+

**Comment by d-fence** [comment]
robodoo check

**Comment by brboi** [comment]
robodoo r+

**Pull request closed without merging**